metadata-ingestion/src/datahub/ingestion/source/mongodb.py

```diff
@@ -282,7 +282,8 @@ def __init__(self, ctx: PipelineContext, config: MongoDBConfig):
             **self.config.options,
         }
 
-        self.mongo_client = pymongo.MongoClient(self.config.connect_uri, **options)  # type: ignore
+        self.mongo_client = pymongo.MongoClient(self.config.connect_uri, datetime_conversion='DATETIME_AUTO',
```

Comment:
let's add a link to https://pymongo.readthedocs.io/en/stable/examples/datetimes.html#handling-out-of-range-datetimes as a comment

Reply:
I took a good look at this link, it was indeed my problem, and I'm very sorry that it was my negligence